server/src/core/link-generator/types.ts

```diff
@@ -8,6 +8,6 @@ export interface DonateLinkArgs {
 }
 
 export interface LinkGeneratorService {
-  getUserDonateLink(user: User, amount: number): string;
+  getUserDonateLink(user: User, amount: number): Promise<string>;
   getDonateLink(args: DonateLinkArgs): string;
```

Comment:
Maybe the `getDonateLink` could have an extra arg of whether the link should be shortened or not.